src/compiler/scanner.ts

```diff
@@ -1226,17 +1243,60 @@ namespace ts {
                 case CharacterCodes.doubleQuote:
                     return "\"";
                 case CharacterCodes.u:
+                    if (isTaggedTemplate) {
+                        // '\u'
+                        if (pos < end && !isHexDigit(text.charCodeAt(pos)) && text.charCodeAt(pos) !== CharacterCodes.openBrace) {
+                            tokenFlags |= TokenFlags.ContainsInvalidEscape;
+                            return text.substring(start, pos);
+                        }
+
+                        // '\u0' or '\u00' or '\u000'
+                        for (let i = 0; i < 3; i++) {

```

Comment:
Maybe call it `escapePos` and initialize it to `pos` (bounded against `pos + 3`) so you don't have to write `pos + i` over and over.